### PR TITLE
Add fx4127 [v101] Integrate search bar position with Nimbus

### PR DESCRIPTION
**Pull request by electricRGB**
[JIRA](https://mozilla-hub.atlassian.net/browse/FXIOS-4127) reference.

### CODE REVIEW

File: Client/Frontend/Settings/SearchBar/SearchBarSettingsViewModel.swift [hunk 1]
```diff
@@ -79,7 +77,8 @@ private extension SearchBarSettingsViewModel {
 
     /// New user defaults to bottom search bar, existing users keep their existing search bar position
     func getDefaultSearchPosition() -> SearchBarPosition {
-        return InstallType.get() == .fresh ? .bottom : .top
+        let nimbusPosition = nimbusBottomSearchBar.getDefaultPosition()
+        return InstallType.get() == .fresh ? nimbusPosition : .top
     }
 
     func saveSearchBarPosition(_ searchBarPosition: SearchBarPosition, shouldNotify: Bool = true) {
```
Comment (lmarceau) [inline Client/Frontend/Settings/SearchBar/SearchBarSettingsViewModel.swift:84]: shouldNotify param could be removed then I think

File: Client/Frontend/Settings/SearchBar/SearchBarSettingsViewModel.swift [hunk 2]
```diff
@@ -46,8 +46,6 @@ final class SearchBarSettingsViewModel {
     var searchBarPosition: SearchBarPosition {
         guard let raw = prefs.stringForKey(PrefsKeys.KeySearchBarPosition) else {
             let defaultPosition = getDefaultSearchPosition()
-            // Do not notify if it's the default position being saved
-            saveSearchBarPosition(defaultPosition, shouldNotify: false)
             return defaultPosition
```
Comment (lmarceau) [inline Client/Frontend/Settings/SearchBar/SearchBarSettingsViewModel.swift:49]: nit
```
return getDefaultSearchPosition()
```

File: Client/Nimbus/NimbusManager.swift [hunk 3]
```diff
@@ -4,26 +4,50 @@
...
-    var sponsoredTileLayer: SponsoredTileLayer
+    var sponsoredTileLayer: NimbusSponsoredTileLayer
+    var bottomSearchBarLayer: NimbusBottomBarLayer
 
     init(with featureFlagLayer: NimbusFeatureFlagLayer = NimbusFeatureFlagLayer(),
-         sponsoredTileLayer: SponsoredTileLayer = SponsoredTileLayer()) {
+         sponsoredTileLayer: NimbusSponsoredTileLayer = NimbusSponsoredTileLayer(),
+         bottomSearchBarLayer: NimbusBottomBarLayer = NimbusBottomBarLayer()
+
+    ) {
```
Comment (lmarceau) [inline Client/Nimbus/NimbusManager.swift:48]:
```suggestion
         bottomSearchBarLayer: NimbusBottomBarLayer = NimbusBottomBarLayer()
    ) {
```

File: Client/Frontend/Settings/SearchBar/SearchBarSettingsViewModel.swift [hunk 4]
```diff
@@ -79,16 +76,16 @@ private extension SearchBarSettingsViewModel {
 
     /// New user defaults to bottom search bar, existing users keep their existing search bar position
```
Comment (lmarceau) [inline Client/Frontend/Settings/SearchBar/SearchBarSettingsViewModel.swift:77]: We can adjust the comment here for this function I think
`New user defaults to bottom search bar` isn't correct anymore in all cases

**Review by lmarceau — APPROVED** [review]

**Comment by electricRGB** [comment]
Screen Shot 2022-05-16 at 2 06 33 PM

**Pull request merged**